web/src/components/product/ProductPage.jsx

```diff
@@ -0,0 +1,435 @@
+
+import { _ } from "~/i18n";
+import { BUSY } from "~/client/status";
+import { If, Page, Popup, Section } from "~/components/core";
+import { noop, useCancellablePromise } from "~/utils";
+import { ProductRegistrationForm, ProductSelector } from "~/components/product";
+import { useInstallerClient } from "~/context/installer";
+import { useProduct } from "~/context/product";
+
+/**
+ * Popup for selecting a product.
+ * @component
+ *
+ * @param {object} props
+ * @param {boolean} props.isOpen
+ * @param {function} props.onFinish - Callback to be called when the product is correctly selected.
+ * @param {function} props.onCancel - Callback to be called when the product selection is canceled.
+ */
+const ProductSelectionPopup = ({ isOpen = false, onFinish = noop, onCancel = noop }) => {
+  const { manager, software } = useInstallerClient();
+  const { products, selectedProduct } = useProduct();
+  const [newProductId, setNewProductId] = useState(selectedProduct?.id);
+
+  const onSubmit = async (e) => {
+    e.preventDefault();
+
+    if (newProductId !== selectedProduct?.id) {
+      await software.product.select(newProductId);
+      manager.startProbing();
+    }
+
+    onFinish();
+  };
+
+  return (
+    <Popup
+      title={_("Choose a product")}
+      isOpen={isOpen}
+    >
+      <Form id="productSelectionForm" onSubmit={onSubmit}>
+        <ProductSelector value={newProductId} products={products} onChange={setNewProductId} />
+      </Form>
+      <Popup.Actions>
+        <Popup.Confirm form="productSelectionForm" type="submit">
+          {_("Accept")}
+        </Popup.Confirm>
+        <Popup.Cancel onClick={onCancel} />
+      </Popup.Actions>
+    </Popup>
+  );
+};
+
+/**
+ * Popup for registering a product.
+ * @component
+ *
+ * @param {object} props
+ * @param {boolean} props.isOpen
+ * @param {function} props.onFinish - Callback to be called when the product is correctly
+ *  registered.
+ * @param {function} props.onCancel - Callback to be called when the product registration is
+ *  canceled.
+ */
+const ProductRegistrationPopup = ({
+  isOpen = false,
+  onFinish = noop,
+  onCancel: onCancelProp = noop
+}) => {
+  const { software } = useInstallerClient();
+  const { selectedProduct } = useProduct();
+  const [isLoading, setIsLoading] = useState(false);
+  const [isFormValid, setIsFormValid] = useState(true);
+  const [error, setError] = useState();
+
+  const onSubmit = async ({ code, email }) => {
+    setIsLoading(true);
+    const result = await software.product.register(code, email);
+    setIsLoading(false);
+    if (result.success) {
+      software.probe();
+      onFinish();
+    } else {
+      setError(result.message);
+    }
+  };
+
+  const onCancel = () => {
+    setError(null);
+    onCancelProp();
+  };
+
+  const isDisabled = isLoading || !isFormValid;
+
+  return (
+    <Popup
+      title={sprintf(_("Register %s"), selectedProduct.name)}
+      isOpen={isOpen}
+    >
+      <If
```

Comment:
This is what I meant in previous comment. This is where I'd use the `<If>` component, as you did it.